---
---
<!DOCTYPE html>
<html>
{% include header.html %}
  <body>

    <header>
      <div class="container">
        <h1>PyBamView</h1>
        <h2>Browser based application for viewing bam alignments</h2>
    {% include menu.html %}<br>
      </div>
    </header>
    <div class="container">
      <section id="main_content">
        <h1>
	  <a name="pybamview" class="anchor" href="#pybamview"><span class="octicon octicon-link"></span></a>faq</h1>

	<h2>Bam files</h2>
	<ul>
	  <li><strong>How does PyBamView determine which samples are present in BAM files?</strong>
	    <p>
	      PyBamView uses read groups to determine which samples to display. Any read group with the same sample (SM) field will be treated as coming from the same sample, even if reads for that sample span multiple bam files. <a href="http://gatkforums.broadinstitute.org/discussion/1317/collected-faqs-about-bam-files">This page on the GATK website</a> answers general FAQ about formatting BAM files.
	  </li>
	  <li><strong>Are clipped bases displayed?</strong>
	    <p>
	    No, all clipped bases (soft and hard clipped) are hidden from display.
	  </li>
	</ul>
	<h2>Reference genome</h2>
	<ul>
	  <li><strong>The reference genome appears as all "N"'s in the browser, even though I provided PyBamView with a reference fasta file.</strong>
	    <p>
	      PyBamView looks up the sequence of the reference genome using the chromosome name and genomic coordinate provided. Therefore, if you look up "chr1:10000", it will only find the reference sequence if there is a chromosome in the reference file with the exact header ">chr1". If the headings of the entries in the reference fasta file look something like ">1" or "1 dna:chrosome 1", PyBamView will not recognize the reference sequence appropriately.
	  </li>
	</ul>
	<h2>Targets</h2>
	<ul>
	  <li><strong>How can I load a set of target loci to pybamview?</strong>
	    <p>
	      If you want to view alignments for a specific set of targeted loci, pybamview has an option to load a bed file with these targets using the --targets option. Then a dropdown box will appear on the alignments page allowing you to jump to different target regions. There is an example targets file in examples/example_targets.bed containing Y-STR and CODIS loci. This file should have 4 tab-delimited columns: chrom, start, end, and name of the marker. The name will be used in the drop down box. 
	</ul>
      </section>
    </div>

    
  </body>
</html>
