---
---
<!DOCTYPE html>
<html>
{% include header.html %}
  <body>

    <header>
      <div class="container">
        <h1>PyBamView</h1>
        <h2>Browser based application for viewing bam alignments</h2>
    {% include menu.html %}<br>
      </div>
    </header>
    <div class="container">
      <section id="main_content">
        <h1>
	  <a name="pybamview" class="anchor" href="#pybamview"><span class="octicon octicon-link"></span></a>faq</h1>
	<h2>General</h2>
	<ul>
	  <li><strong>How many samples/reads can PyBamView visualize at once?</strong>
	    <p>
	      Theoretically, there is no limit to the number of samples or reads PyBamView can visualize. However, visualizing more than 5 or so samples can become cumbersome, both because the time to load each view increases and because it is hard to fit alignments for that many samples on one page.
	    <p>
	      Similarly, PyBamView might have trouble if you try to view too many reads at once. If you have extremely high coverage sequencing of targeted loci (e.g. more than 50-100x), you may want to consider downsampling your reads or removing PCR duplicates before visualizing to improve performance and visualization.
	  </li>
	</ul>
	<h2>Bam files</h2>
	<ul>
	  <li><strong>How does PyBamView determine which samples are present in BAM files?</strong>
	    <p>
	      PyBamView uses read groups to determine which samples to display. Any read group with the same sample (SM) field will be treated as coming from the same sample, even if reads for that sample span multiple bam files. Note that not all aligners automatically annotate read groups. You can use Picard's <a href="http://picard.sourceforge.net/command-line-overview.shtml#AddOrReplaceReadGroups">AddOrReplaceReadGroups</a> tool to add read group information.
	      <p>
		<a href="http://gatkforums.broadinstitute.org/discussion/1317/collected-faqs-about-bam-files">This page on the GATK website</a> answers general FAQ about formatting BAM files.
	  </li>
	  <li><strong>Are clipped bases displayed?</strong>
	    <p>
	      No, all clipped bases (soft and hard clipped) are hidden from display.
	  </li>
	  <li><strong>Why does my called INDEL not show up in pybamview?</strong>
	    <p>
	      In some cases it is possible that INDELs called by a genotyping algorithm will not show up in pybamview. This may happen if, for example, one uses GATK's HaplotypeCaller (HC) to call INDELs. HC performs local re-assembly of reads, but normally outputs only VCF. Therefore, it may end up calling an INDEL that was not picked up by the aligner, and therefore will not appear in the original BAM.
	    <p>
	      One solution is to use HaplotypeCaller's --bamOutput option, and load the resulting BAM rather than the original. (Thanks to <a href="https://twitter.com/roy_ronen">@roy_ronen</a> for pointing this out).
	  </li>
	  <li><strong>What size BAM files can PyBamView handle?</strong>
	    <p>
	      PyBamView can handle quite large BAM files (e.g. many gigabytes) since it only pulls out the reads it needs for the specific location you are viewing. However, for experiments with extremely high coverage at a small set of targeted regions, PyBamView can become very slow and inconvenient. PyBamView should handle most standard whole genome and whole exome experiments.
	  </li>
	</ul>
	<h2>Reference genome</h2>
	<ul>
	  <li><strong>The reference genome appears as all "N"'s in the browser, even though I provided PyBamView with a reference fasta file.</strong>
	    <p>
	      PyBamView looks up the sequence of the reference genome using the chromosome name and genomic coordinate provided. Therefore, if you look up "chr1:10000", it will only find the reference sequence if there is a chromosome in the reference file with the exact header "&#62;chr1". If the headings of the entries in the reference fasta file look something like "&#62;1" or "1 dna:chrosome 1", PyBamView will not recognize the reference sequence appropriately.
	  </li>
	</ul>
	<h2>Web browser display</h2>
	<ul>
	  <li><strong>How can I run PyBamView from one computer and view alignments in a web browser from another computer?</strong>
	    <p>
	      Run PyBamView normally, but specify "--ip 0.0.0.0 --no-browser". Then determine the hostname or IP address of the computer you are running PyBamView from (e.g. you can get the hostname using the command "hostname -f"). Then, open a web browser on another computer and navigate to "http://HOSTNAME:5000" (assuming you're using the default port 5000). Note this requires that the specified port is open for incoming http traffic. See <a href="serve-pybamview-remotely.html">this page</a> for more details.
	  </li>
	</ul>
	<h2>Targets</h2>
	<ul>
	  <li><strong>How can I load a set of target loci to pybamview?</strong>
	    <p>
	      If you want to view alignments for a specific set of targeted loci, pybamview has an option to load a bed file with these targets using the --targets option. Then a dropdown box will appear on the alignments page allowing you to jump to different target regions. There is an example targets file in examples/example_targets.bed containing Y-STR and CODIS loci. This file should have 4 tab-delimited columns: chrom, start, end, and name of the marker. The name will be used in the drop down box. 
	</ul>
      </section>
    </div>

    
  </body>
</html>
