---
---
<!DOCTYPE html>
<html>
{% include header.html %}
  <body>

    <header>
      <div class="container">
        <h1>PyBamView</h1>
        <h2>Browser-based application for viewing bam alignments</h2>
    {% include menu.html %}<br>
      </div>
    </header>
    <div class="container">
      <section id="main_content">
        <h1>
<a name="pybamview" class="anchor" href="#pybamview"><span class="octicon octicon-link"></span></a>pybamview</h1>
<p>Browser-based application for viewing bam alignments</p>
PyBamView is a python-based application that allows you to view BAM sequence alignments through your web browser. It was inspired by the <a href="http://samtools.sourceforge.net/samtools.shtml">samtools tview</a> tool, and was motivated by the need to better visualize alignments with complex insertions/deletions across multiple samples, which is not handled well by current broswers such as IGV and the UCSC Genome Browser.
<img src="images/pybamview_example.png">
PyBamView is run locally, without the need to upload large alignment/genome files to external sites, and provides text alignments that are fast and easy to navigate. Take a look at the <a href="http://pybamview.melissagymrek.com/bamview?bamfiles=%2Fmnt%2FHG00096.chrom20.ILLUMINA.bwa.GBR.low_coverage.20120522.bam&bamfiles=%2Fmnt%2FHG00121.chrom20.ILLUMINA.bwa.GBR.low_coverage.20130415.bam&samples=HG00096&samples=HG00121&zoomlevel=1&region=20%3A1400000">example</a> and <a href="http://melissagymrek.com/python/2014/01/29/pybamview.html">my blog post</a> to learn more.
<p><em>*Your feedback is welcome! Email mgymrek AT mit DOT edu or tweet at <a href="https://twitter.com/mgymrek">@mgymrek</a>*</em></p>
      </section>
    </div>

    
  </body>
</html>
