---
---
<!DOCTYPE html>
<html>
{% include header.html %}
  <body>

    <header>
      <div class="container">
        <h1>PyBamView</h1>
        <h2>Browser based application for viewing bam alignments</h2>
    {% include menu.html %}<br>
      </div>
    </header>
    <div class="container">
      <section id="main_content">
        <h1>
<a name="pybamview" class="anchor" href="#pybamview"><span class="octicon octicon-link"></span></a>pybamview</h1>
<p>Browser based application for viewing bam alignments</p>
<p><em>* This repository is under construction. But you're feedback is welcome! (Email mgymrek AT mit DOT edu or tweet at <a href="https://twitter.com/mgymrek">@mgymrek</a>) *</em></p>
PyBamView is a python based application that allows you to view BAM DNA sequence alignments through your web browser. It was inspired by the <a href="http://samtools.sourceforge.net/samtools.shtml">samtools tview</a> tool, and was motivated by the need to better visualize alignments with complex insertions/deletions across multiple samples, which is not handled well by current broswers such as IGV and the UCSC Genome Browser.
<img src="images/pybamview_example.png">
PyBamView is run locally, without the need to upload large alignment/genome files to external sites, and provides text-based alignments that are fast and easy to navigate. Take a look at the <a href="http://pybamview.melissagymrek.com/?bamfiles=HG00096.chrom20.ILLUMINA.bwa.GBR.low_coverage.20120522.bam&bamfiles=HG00121.chrom20.ILLUMINA.bwa.GBR.low_coverage.20130415.bam&region=20%3A61100">example</a> to learn more.
      </section>
    </div>

    
  </body>
</html>
